src/lang-go.ts

```diff
@@ -563,6 +558,121 @@ function positionParams(doc: sourcegraph.TextDocument, pos: sourcegraph.Position
+                          address.protocol
+                      }\`** is invalid. Only **\`ws:\`** or **\`wss:\`** are valid.`,
+                      '',
+                  ]),
+            '',
+            `To apply the fix:`,
+            '',
+            `- Correct the value in your [global settings](${
+                sourcegraph.internal.sourcegraphURL
+            }site-admin/global-settings).`,
+            '',
+            'To temporarily disable:',
+            '',
+            `- Comment out **\`go.serverUrl\`** in your [global settings](${
+                sourcegraph.internal.sourcegraphURL
+            }site-admin/global-settings).`,
+        ].join('\n')
+    )
+}
+
+function notifyLanguageServerUnableToConnectToSourcegraph(): void {
+    sourcegraph.internal.updateContext({
+        showError: true,
+        'codeIntel.error.message': 'Error in language server setup. Click to open documentation.',
+        'codeIntel.error.link': 'https://sourcegraph.com/extensions/sourcegraph/go',
+    })
+    sourcegraph.app.activeWindow!.showNotification(
+        [
+            '**❌ Error in Go language server setup**',
+            '',
+            `The Go language server was unable to fetch repository contents from Sourcegraph at:`,
+            '',
+            '```',
+            `"go.sourcegraphUrl": "${sourcegraphURL()}"`,
+            '```',
+            '',
+            'To troubleshoot this, **`docker exec`** into the go-langserver container and run:',
+            '',
+            `- **\`ping ${sourcegraphURL().hostname}\`**`,
+            `- **\`curl ${sourcegraphURL()}\`**`,
+            '',
+            `To apply the fix:`,
+            '',
+            `- Correct the value in your [global settings](${
+                sourcegraph.internal.sourcegraphURL
+            }site-admin/global-settings).`,
+            '',
+            'To temporarily disable:',
+            '',
+            `- Comment out **\`go.serverUrl\`** in your [global settings](${
+                sourcegraph.internal.sourcegraphURL
+            }site-admin/global-settings).`,
+        ].join('\n')
+    )
+}
+
+async function canary(address: URL): Promise<void> {
+    const data = await queryGraphQL('{currentUser{siteAdmin}}')
+    if (!data || !data.currentUser || data.currentUser.siteAdmin === undefined) {
+        console.log(
+            'Failed to determine whether or not the current user is an admin. Check the Network tab to see what went wrong.'
+        )
+        return
+    }
+    if (!data.currentUser.siteAdmin) {
+        // Don't show these notifications to normal users because they don't
+        // have access to global settings, and therefore can't fix this for all
+        // users.
+        return
+    }
+
+    try {
+        await new Promise<void>((resolve, reject) => {
+            let webSocket: WebSocket
+            try {
+                webSocket = new WebSocket(address.href)
+            } catch (e) {
```

Comment:
This error would get caught by the outer `catch`, which has the same handling, so this try/catch appears redundant